src/realm/group_writer.cpp

```diff
@@ -565,14 +580,32 @@ ref_type GroupWriter::write_group()
     // If file has a history and is opened in shared mode, write the new history
     // to the file. If the file has a history, but si not opened in shared mode,
     // discard the history, as it could otherwise be left in an inconsistent state.
-    if (top.size() >= 8) {
-        REALM_ASSERT(top.size() >= 10);
-        // In nonshared mode, history must already have been discarded by GroupWriter constructor.
-        if (ref_type history_ref = top.get_as_ref(8)) {
+    if (top.size() > Group::s_hist_ref_ndx) {
+        if (ref_type history_ref = top.get_as_ref(Group::s_hist_ref_ndx)) {
             Allocator& alloc = top.get_alloc();
             ref_type new_history_ref = Array::write(history_ref, alloc, *this, only_if_modified); // Throws
-            int_fast64_t value_3 = from_ref(new_history_ref);
-            top.set(8, value_3); // Throws
+            top.set(Group::s_hist_ref_ndx, from_ref(new_history_ref));                            // Throws
+        }
+    }
+    if (top.size() > Group::s_evacuation_point_ndx) {
+        ref_type ref = top.get_as_ref(Group::s_evacuation_point_ndx);
+        if (m_evacuation_limit) {
+            REALM_ASSERT(ref);
+            Array arr(m_alloc);
+            arr.init_from_ref(ref);
+            arr.truncate(2);
+
+            arr.set(0, int64_t(m_evacuation_limit));
+            arr.set(1, m_evacuation_progress.empty() ? 1 : 0); // Backoff from scanning
```

Comment:
Is the backoff field necessary? Or could it be implied by reading the number of progress entries?

Reply:
Maybe. But at some point, I had the idea that we should only backoff for a given number of commits. I think it is better to reserve the space for that kind of information.